### PR TITLE
Toggle favorite backend

### PR DESCRIPTION
**Pull request by DDVD233**
Small ticket. Hopefully easy to review/

### CODE REVIEW

File: app_backend/app/routers/favorites.py [hunk 1]
```diff
@@ -34,11 +44,17 @@ async def toggle_favorite(resp: Response, favorite: FavoriteFood):
...
-    existing_food_state = True  # TODO: get food_id's state here
-    if existing_food_state == favorite.state:
+    favorites = user.favorites
+    if (favorite.food_name in favorites) == favorite.state:
         resp.status_code = status.HTTP_200_OK
         return {"msg": "No row is affected."}
-    # TODO: Update food_id here
-    # toggle food_id
+    else:
+        if favorite.state:
```
Comment (RafaelPiloto10) [inline app_backend/app/routers/favorites.py:52]: Not sure if I am understanding this bit correctly, how does it toggle from already favorite to not favorite? It seems like you have a list of favorite food names. I like this -- maybe just get rid of states? 

When toggling from favorite to not favorite remove from the list?

File: app_backend/app/routers/favorites.py [hunk 2]
```diff
@@ -34,11 +44,17 @@ async def toggle_favorite(resp: Response, favorite: FavoriteFood):
...
+            favorites.append(favorite.food_name)
+
+    # Save
+    user.favorites = favorites
+    user.save()
```
Comment (RafaelPiloto10) [inline app_backend/app/routers/favorites.py:59]: add `dry_run` parameter in this method that can be set in the pay_load in tests. We don't want to write data when testing

**Review by RafaelPiloto10 — CHANGES_REQUESTED** [review]

File: app_backend/app/routers/favorites.py (hunk 1 above)
Comment (DDVD233) [reply app_backend/app/routers/favorites.py:52]: Oh, I forgot to remove them from the list. Will do it now.

**Comment by RafaelPiloto10** [comment]
@DDavid233 do you plan to make a separate PR for integrating this with the front end? I think its fine to do it here since its so small. You can use `Auth().isTest()` in your http request on toggle to avoid making the HTTP request when testing as that causes an error

File: app_backend/models/user.py [hunk 3]
```diff
@@ -37,4 +37,5 @@ def get_user_by_email(self, email):
         return user
 
     def save(self):
-        db.db.users.update_one({"_id": self._id}, {"$set": self.dict()}, upsert=True)
+        if not db.test_mode:
```
Comment (RafaelPiloto10) [inline app_backend/models/user.py:40]: I love your clever solutions, lol. Nice one!

**Review by RafaelPiloto10 — COMMENTED** [review]
See recent comment

**Comment by DDVD233** [comment]
@RafaelPiloto10 
Yes, I was actually doing it this morning. I had to modify auth.dart to make it work, but then I realized you modified it too...So while I am currently not opening a PR for it, I have it under frontend-toggle-favorite branch.

**Review by RafaelPiloto10 — APPROVED** [review]
Sounds good! Thanks

**Review by brendacano — APPROVED** [review]
This all looks good to me. I can't think if any way it can be improved.

**Pull request merged**